src/librustdoc/passes/html_tags.rs

```diff
@@ -249,7 +249,31 @@ impl<'a, 'tcx> DocVisitor for InvalidHtmlTagsLinter<'a, 'tcx> {
             let mut is_in_comment = None;
             let mut in_code_block = false;
 
-            let p = Parser::new_ext(&dox, main_body_opts()).into_offset_iter();
+            let link_names = item.link_names(&self.cx.cache);
+
+            let mut replacer = |broken_link: BrokenLink<'_>| {
+                if let Some(link) =
+                    link_names.iter().find(|link| *link.original_text == *broken_link.reference)
```

Comment:
This seems like it might be kind of slow... maybe worth addressing in the future (not in this PR).

Reply:
We will run a perf check just to be sure the impact isn't too big, otherwise it'll have to be fixed in this PR (good catch!).